src/components/2dmap/TwoDimensionalMap.tsx

```diff
@@ -61,35 +74,39 @@ export default class TwoDimensionalMap extends React.Component<ITwoDimensionalMa
    * Lifecycle react method. This method is triggered when the react component is correctly loaded into the dom.
    */
   public componentDidMount() {
-    this.generateMap();
+    const { selectedProduct, products } = this.props.productData;
+    this.generateMap(selectedProduct, products);
+  }
+
+  public componentWillReceiveProps(nextprops: props) {
```

Comment:
I'd like to know why both componentDidMount() and componentWillRecieveProps() are needed when they do the same thing.

Reply:
Without loading the map the first time aroound it wont load until you have performed a search

Reply:
The component renders when its load it, but at that point the data is not there yet, you must therefore tell it to render afrer you have received the data for the map